bitcoin/src/blockdata/transaction.rs

```diff
@@ -779,7 +779,7 @@ pub fn effective_value(
     value: Amount,
 ) -> Option<SignedAmount> {
     let weight = input_weight_prediction.total_weight();
-    let signed_input_fee = fee_rate.to_fee(weight)?.to_signed();
+    let signed_input_fee = fee_rate.to_fee(weight).ok()?.to_signed();
```

Comment:
I think it'd be reasonable to do this and the change of the return type in a single commit.

Reply:
not sure if/when this gets merged so probably gonna hold off on investing more changes in it for now.